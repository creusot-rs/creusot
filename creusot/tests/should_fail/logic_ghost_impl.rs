extern crate creusot_contracts;
use creusot_contracts::*;

trait T {
    #[logic]
    fn f();
}

impl T for () {
    #[ghost]
    fn f() {
        ()
    }
}
