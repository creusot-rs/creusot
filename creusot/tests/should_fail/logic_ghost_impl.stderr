Logic != Ghost
error[creusot]: Expected `f` to be a ghost function as specified by the trait declaration
  --> logic_ghost_impl.rs:11:5
   |
11 |     fn f() {
   |     ^^^^^^

error: aborting due to previous error

